service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderControllerMidEventTest.java

```diff
@@ -598,4 +603,37 @@ private DocumentReference expectedDraftDocument() {
             return childElement;
         }
     }
+
+    @Nested
+    class PopulateEPOTypeAndAddressMidEvent {
+
+        private final String callbackType = "populate-epo-parameters";
+
+        @Test
+        void shouldPrePopulateEpoOrderFieldsIfPresentInCaseData() {
+            AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(
+                buildCaseData(REMOVE_TO_ACCOMMODATION), callbackType);
+
+            CaseData caseData = extractCaseData(callbackResponse);
+
+            assertThat(caseData.getOrders().getEpoType().equals(REMOVE_TO_ACCOMMODATION));
+        }
+
+        @Test
+        void shouldPrePopulateAddressFieldsIfPresentInCaseData() {
+            AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(
+                buildCaseData(PREVENT_REMOVAL), callbackType);
+
+            CaseData caseData = extractCaseData(callbackResponse);
+
+            assertThat(caseData.getOrders().getEpoType().equals(PREVENT_REMOVAL));
+        }
+
+        private CaseData buildCaseData(EPOType epoType) {
+            return CaseData.builder()
+                .orders(Orders.builder()
```

Comment:
might also be worth adding some validation tests here to check that errors are being returned when invalid address is added

Reply:
are you referring to uk.gov.hmcts.reform.fpl.controllers.ValidateOrderControllerTest#shouldReturnErrorsWhenEPOTypeIsPreventRemovalButAddressIsIncomplete ?

Reply:
agreed with Toireasa

Reply:
oh yes, we test this elsewhere. all good then. :)